processing/src/main/java/org/apache/druid/segment/filter/BoundFilter.java

```diff
@@ -75,71 +76,107 @@ public BitmapColumnIndex getBitmapColumnIndex(ColumnIndexSelector selector)
     if (!Filters.checkFilterTuningUseIndex(boundDimFilter.getDimension(), selector, filterTuning)) {
       return null;
     }
-    if (supportShortCircuit()) {
+    if (supportStringShortCircuit()) {
       final ColumnIndexSupplier indexSupplier = selector.getIndexSupplier(boundDimFilter.getDimension());
       if (indexSupplier == null) {
         return Filters.makeNullIndex(doesMatchNull(), selector);
       }
       final LexicographicalRangeIndex rangeIndex = indexSupplier.as(LexicographicalRangeIndex.class);
-      if (rangeIndex == null) {
-        // column
-        return null;
+      if (rangeIndex != null) {
+        final BitmapColumnIndex rangeBitmaps = rangeIndex.forRange(
+            boundDimFilter.getLower(),
+            boundDimFilter.isLowerStrict(),
+            boundDimFilter.getUpper(),
+            boundDimFilter.isUpperStrict()
+        );
+        // preserve sad backwards compatible behavior where bound filter matches 'null' if the lower bound is not set
```

Comment:
I think this can be refactor outside of the if as it is common to both the String and the Numeric

Reply:
they just look really similar right now, but the range indexes don't share a common interface currently so isn't super easy